### PR TITLE
Removed azuread dependency wherever applicable for AAD Diagnostic settings

### PR DESCRIPTION
**Pull request by PriyankaRanganath**
…ting feature

This PR is for removing azuread dependency from azurerm wherever applicable. I have implemented azure AD diagnostic settings feature in azuread provider. But I am unable to test them due to cyclic dependency between azurerm and azuread providers.
To avoid that, this PR has removed azuread dependencies wherever applicable and provided equivalent functions for the same.

### CODE REVIEW

**Comment by PriyankaRanganath** [comment]
Hi @manicminer , can you please review the PR along with [this ](https://github.com/terraform-providers/terraform-provider-azurerm/pull/8581) too ?
They are sitting idly for past 4-5 days and waiting for review.

File: azurerm/internal/acceptance/data.go [hunk 1]
```diff
@@ -64,7 +65,7 @@ func BuildTestData(t *testing.T, resourceType string, resourceLabel string) Test
 	}
 
 	testData := TestData{
-		RandomInteger:   tf.AccRandTimeInt(),
+		RandomInteger:   AccRandTimeInt(),
```
Comment (katbyte) [inline azurerm/internal/acceptance/data.go:68]: then this can go back to
```suggestion
		RandomInteger:   tf.AccRandTimeInt(),
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for fixing this mistake @PriyankaRanganath! aside from one comment i've left inline this looks good

File: azurerm/internal/acceptance/data.go [hunk 2]
```diff
@@ -5,11 +5,12 @@ import (
...
 	"os"
 	"strconv"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
-	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/tf"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
```
[the remark this answers is not in the document]
Comment (PriyankaRanganath) [reply azurerm/internal/acceptance/data.go:14]: Hi @katbyte, I already tried that. If I use azurerm tf randint function, lot of acceptance tests fails due to too much length restrictions in names or other properties. That's why I supplied a new function same as the one used in azuread tf provider

[the remark this answers is not in the document]
Comment (katbyte) [reply azurerm/internal/acceptance/data.go:14]: That shouldn't be the case? the functions are almost identical? regardless can you swap it and we'll handle it on our end if there is an issue with it

**Comment by PriyankaRanganath** [comment]
Hi @katbyte , Have removed the extra randint function and added reference to azurerm tf randint function as requested.

**Review by katbyte — APPROVED** [review]
Thanks @PriyankaRanganath - LGTM 👍

**Pull request merged**